linera-execution/src/execution_state_actor.rs

```diff
@@ -387,13 +387,8 @@ where
                 http_responses_are_oracle_responses,
                 callback,
             } => {
-                let response = if let Some(response) =
-                    self.txn_tracker.next_replayed_oracle_response()?
-                {
-                    match response {
-                        OracleResponse::Http(response) => response,
-                        _ => return Err(ExecutionError::OracleResponseMismatch),
-                    }
+                let maybe_response = if self.txn_tracker.is_replaying() {
```

Comment:
shouldn't this be a negation?
```suggestion
                let maybe_response = !if self.txn_tracker.is_replaying() {
```

Reply:
This is more confusing than before IMO. 

Previously, when `next_replayed_oracle_response` returned `Some(Http(response))` we'd simply add that to `add_oracle_response`.

Now the `add_oracle_response` has new logic included that interprets `None` as "we should use a next entry from the oracle responses.

Maybe it'd be bit clearer if you didn't try to have single `let response` here and special case in the `if { ... }` and `else { ... }` clauses?

Reply:
Yes, maybe it's not much of an improvement after all. Let's do the removal part first: https://github.com/linera-io/linera-protocol/pull/4467

It's not in all call sites as simple as putting the response in the callback, so I think what you're suggesting would basically be the approach from before.

The case I'm worried about is that in one of the many places where we use oracles we forget to add the response in one of the two cases (or both).